bond/bond_test.go

```diff
@@ -326,64 +288,177 @@ var _ = Describe("bond plugin", func() {
+			args := &skel.CmdArgs{
+				ContainerID: "dummy",
+				Netns:       podNS.Path(),
+				IfName:      IfName,
+				StdinData:   []byte(fmt.Sprintf(Config, "0.3.1", ActiveBackupMode, strconv.FormatBool(linksInContainer), bondMTU)),
+			}
+			By("creating the plugin")
+			_, _, err := testutils.CmdAddWithArgs(args, func() error {
+				return cmdAdd(args)
+			})
+			Expect(err).To(HaveOccurred())
+		},
+			Entry("Bond MTU is bigger then one of links MTU", "1200"),
+			Entry("Bond MTU is bigger then all of links MTU", "900"),
+		)
+	})
+	When("links are in host`s network namespace at initial state (meaning linksInContainer is false)", func() {
+		BeforeEach(func() {
+			linksInContainer = false
+			linkAttrs = []netlink.LinkAttrs{
+				{Name: Slave1},
+				{Name: Slave2},
+			}
+		})
+		It("verifies a plugin is added and deleted correctly ", func() {
+			args := &skel.CmdArgs{
+				ContainerID: "dummy",
+				Netns:       podNS.Path(),
+				IfName:      IfName,
+				StdinData:   []byte(fmt.Sprintf(Config, "0.3.1", ActiveBackupMode, strconv.FormatBool(linksInContainer), DefaultMTU)),
+			}
 
 			By("creating the plugin")
 			r, _, err := testutils.CmdAddWithArgs(args, func() error {
 				return cmdAdd(args)
 			})
 			Expect(err).NotTo(HaveOccurred())
+			By("validating the returned result is correct")
+			checkAddReturnResult(&r, IfName)
 
-			By("checking the bond was created")
-			result, err := types100.GetResult(r)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(len(result.Interfaces)).To(Equal(1))
-			Expect(result.Interfaces[0].Name).To(Equal(IfName))
+			_ = podNS.Do(func(ns.NetNS) error {
+				defer GinkgoRecover()
+				By("validating the bond interface is configured correctly")
+				link, err := netlink.LinkByName(IfName)
+				Expect(err).NotTo(HaveOccurred())
+				validateBondIFConf(link, 1400, ActiveBackupMode, 100)
 
-			By("duplicating the macs on the slaves")
-			err = netlink.LinkSetHardwareAddr(slave2, slave1.Attrs().HardwareAddr)
-			Expect(err).NotTo(HaveOccurred())
+				By("validating the bond slaves are configured correctly")
+				validateBondSlavesConf(link, Slaves)
+				return nil
+			})
 
-			By("deleting the plugin")
-			err = testutils.CmdDel(podNS.Path(),
-				args.ContainerID, "", func() error { return cmdDel(args) })
+			By("validating the bond interface is deleted correctly")
+			err = testutils.CmdDelWithArgs(args, func() error {
+				return cmdDel(args)
+			})
 			Expect(err).NotTo(HaveOccurred())
 
-			By("validating the macs are not duplicated")
-			slave1, err = netlink.LinkByName(Slave1)
+			By("Checking that links are not in pod namespace")
+
+			_ = podNS.Do(func(ns.NetNS) error {
+				defer GinkgoRecover()
+				for _, slaveName := range Slaves {
+					_, err := netlink.LinkByName(slaveName)
+					Expect(err).To(HaveOccurred())
+				}
+				return nil
+			})
+
+			By("Checking that links are in host namespace")
```

Comment:
you can use validateBondSlavesConf or not?

Reply:
not really, the validateBondSlavesConf function validates that the slaves are configured as the bond - in this case i only want to check that the links are  in the init namespace (back to the state they were before the bond), so this function does not match here even if they have some Similar parts.